Update 2018

steffilazerte opened the pull request:
Full update to use new data from new sources and to streamline some of the processes:

**Changes to methodology:**
- Script automatically downloads all data (no raw data needs to be stored locally)
- Each script is self-contained. That is, that each script loads the packages and the data required and saves the output data at the end.
- Mapping code moved to visualization script to keep all figure creation together (ggmap script omitted)

**Date related changes:**
- Includes wells active as of 2008-01-01
- I've updated all references to dates including copyright. The only place I haven't is in the links to 2014 methodology (as that would break the links)

**Rmd file changes:**
- Under pdf_document, Rmd file runs and creates the output as expected. Some of the figures may require tweaking, as the dates on the axes are starting to run into each other. This may not be a problem with the right theme?
- Rmd file now uses the summary figures saved from the visualizations script rather than recreating them again (individual well figures are still created in the Rmd file, though)
- There are no "Station Names" in the data that I could find, so I've used Observation Well number alone in the Rmd file appendix (Let me know if I missed them)

**Things to note:**
- Some of the wells show a convergence code 52 when calculating the trends (I've made a list of them in the clean script, not sure how to proceed with these). At least one (302) shows a Large rate of decline, so this should probably be assessed.

steffilazerte committed:
Remove old data

steffilazerte committed:
Consolidate figures

steffilazerte committed:
Main update

steffilazerte committed:
Just use list, as some will be empty anyway

ateucher committed:
Move bcmaps to CRAN package list

ateucher committed:
Make tmp dir

ateucher commented on 01_clean.R:
```diff
@@ -1,178 +1,136 @@
-# Combine the list of data frames into a single long data frame. This data frame 
-# can then be converted to annual data and analysed.
-monthlywells.ts <- rbind_all(monthlyts_strip)
+wells_ts_trimmed <- mutate(wells_ts, 
```
I have been thinking about the trimming of leading/trailing missing values, and wondered if that should be done before interpolation (so we're not trying to interpolate in really sparse areas at the beginning and ends of the time series). I pushed to a branch in `bcgroundwater` where I incorporated this into `makeWellTS()`: https://github.com/bcgov/bcgroundwater/commit/80026000a0bc5954ec07c3c08fc6e2002bc03139. Then we can omit this part (lines 111:115) and change line `06 to:

```r
wells_ts <- mutate(wells_month, data = map(data, ~{
  makeWellTS(.x, trim = TRUE, head = 0.1, tail = 0.9, n_consec = 4)
}))
```
What do you think? It does work, and there is one less convergence code 52.

steffilazerte replied:
Good point. I think this would work well: it makes sense statistically, it simplifies the code, and it might even speed things up a bit. Also a nice option to include in `bcgroundwater`.

I guess it would be important then to make a note of the `bcgroundwater` version required in the preamble.

ateucher committed:
Change by argument to byID in gwlZypTest call

ateucher committed:
Tweak yaml header for print_ver

ateucher reviewed (commented):
@steffilazerte this looks really great! Just a couple of tweaks I made, otherwise it all runs great. I have one comment/question inline below. We won't merge until we're ready to publish though, so this PR will sit for a while

stephhazlitt committed:
update documentation

stephhazlitt committed:
update documentation

stephhazlitt committed:
update gitignore

stephhazlitt committed:
update README

stephhazlitt committed:
update README

stephhazlitt committed:
add licence header

stephhazlitt committed:
update rproj file name

stephhazlitt committed:
add apache badge

stephhazlitt committed:
fix apache badge

stephhazlitt committed:
capitalize; title edits

stephhazlitt committed:
fix licence header; text edits

stephhazlitt committed:
update text

stephhazlitt committed:
update template and urls

stephhazlitt committed:
update gitignore

stephhazlitt committed:
rename rmd

stephhazlitt committed:
fix typo

stephhazlitt committed:
update documentation

stephhazlitt committed:
add in methods

stephhazlitt committed:
remove separate methods rmd

stephhazlitt committed:
update makeWellTS arguments; remove old trim code

stephhazlitt committed:
fix sigfigs in table; move methods

stephhazlitt committed:
text edits

stephhazlitt committed:
update with bcgroundwater changes

stephhazlitt committed:
plot design tweaks

stephhazlitt committed:
text now black

stephhazlitt committed:
cache=FALSE; text black

stephhazlitt committed:
move save()

stephhazlitt committed:
tweak designs

stephhazlitt committed:
no margin

stephhazlitt committed:
add load

stephhazlitt committed:
update to work with bcgroundwater v1.0

stephhazlitt committed:
tweaking theme()

stephhazlitt committed:
use png_retina

stephhazlitt committed:
theme changes

stephhazlitt committed:
start compare years

stephhazlitt committed:
methods formatting

stephhazlitt committed:
comparing results among years

stephhazlitt committed:
save data file to tmp

stephhazlitt committed:
add well metadata to comparison

stephhazlitt committed:
reduce PDF filesize

stephhazlitt committed:
add dev

stephhazlitt committed:
fix date typo

stephhazlitt committed:
define categories up front; tweak xtable

stephhazlitt committed:
update and add categories

stephhazlitt committed:
comment file.copy

stephhazlitt committed:
require ggmap dev version

stephhazlitt committed:
add region to well titles

stephhazlitt committed:
filter where n month values < 12

stephhazlitt committed:
alt source for metadata

stephhazlitt committed:
aquifer type summary

stephhazlitt committed:
update comments

stephhazlitt committed:
filter historic wells

stephhazlitt committed:
print map

stephhazlitt committed:
overlay sf object on ggmap

stephhazlitt committed:
map line size smaller

stephhazlitt committed:
add header script

stephhazlitt committed:
restructure load and clean code

stephhazlitt committed:
rename analysis file

stephhazlitt committed:
add packages

stephhazlitt committed:
restructure laod & clean code

stephhazlitt committed:
filter obs_wells in load.R

stephhazlitt committed:
update comments

stephhazlitt committed:
remove deprecated code

stephhazlitt committed:
conditional source

stephhazlitt committed:
update comments

stephhazlitt committed:
finish script restructure

stephhazlitt committed:
update README

stephhazlitt committed:
scratch folder; compare script

stephhazlitt committed:
add glue and subfolder

stephhazlitt committed:
write out clean data csv

stephhazlitt committed:
write out metadata csv

stephhazlitt committed:
restructure

stephhazlitt committed:
move write.table to 03

stephhazlitt committed:
tmp outputs

stephhazlitt committed:
use map object from RData

stephhazlitt committed:
remove ggmap

stephhazlitt committed:
comment

stephhazlitt committed:
remove depth and type details

stephhazlitt committed:
use .packages

co-authored by: sam.albers@gov.bc.ca

stephhazlitt committed:
use .header_sourced

co-authored by: andy.teucher@gov.bc.ca

stephhazlitt committed:
move packages to setup

ateucher committed:
Document use of API key for ggmap

ateucher committed:
Ensure dev version of ggmap is installed

stephhazlitt committed:
update prov-level plots

stephhazlitt committed:
Merge branch 'update-2018' of https://github.com/bcgov/groundwater-levels-indicator into update-2018

stephhazlitt committed:
add patchwork

co-authored by: andy.teucher@gov.bc.ca

stephhazlitt committed:
update prov summaries; closes #11

stephhazlitt committed:
update prov summary objects

stephhazlitt committed:
charts for each nr region; closes #12

steffilazerte committed:
Creat individual plots in 04_output, not in Rmd print version
- Add labels
- Only run ggmaps if specified

steffilazerte committed:
Start using plot objects (not creating plots directly)

steffilazerte committed:
Create leaflet_map first draft
- Wells and Regions
- Tool tips
- Well popups drafted

stephhazlitt reviewed (commented):
Nice work refactoring & shifting the individual well trend plot code. I cannot render the first draft of the leaflet map without `create_popup_groundwater()`, but maybe I am jumping the queue re: reviewing that piece?

steffilazerte commented:
You'll need an update to envreportutils to my forked version (https://github.com/steffilazerte/envreportutils/commit/e27e0c856f89d3d8f94f5be63da8dd8b464d2ad6) that I have yet to create a pull request for (and just pushed this morning).

I'm struggling a bit to decide whether `envreportutils` is actually the right place for these files. These functions, while based on the CAAQs functions, are specific to this map. Further, they aren't required by any other repositories (unlike the CAAQS functions which were used by two different indicators in two different repositories).

So that being said, I'm not sure if I should keep them in `envreportutils` or not.

Feel free to take a look, but this is definitely a draft at this point!

steffilazerte committed:
Remove new line from regional plot titles

steffilazerte committed:
Uncomment month plot saving

steffilazerte committed:
Group all well operations together

steffilazerte committed:
Sort wells so decreasing trends are on top

steffilazerte committed:
Create regional popups and add legend

steffilazerte committed:
Add colours to wells for badges and organize to top

steffilazerte committed:
Add java tweak to re-centre popup with autopan

steffilazerte committed:
Replace java tweak with envreportutils function

steffilazerte commented:
Okay, I'm liking this version better!

I've opted with a look more like the CAAQS indicators, but let me know if you'd prefer to omit the badge. 
I also figured out how to force the popup to pan properly on the first time it's opened (a bit of java magic, that I think would also be a good addition to the CAAQs indicators. I added the function the the envreportutils package).

The regional popups are a bit bland, but I'm not sure what else to add/change.

The functions in [steffilazerte/envreportutils](http://github.com/steffilazerte/envreportutils) are actually a bit more generic now, so maybe not a bad place to keep them after all. However, I still haven't gotten around to updated the CAAQS functions, and I'd like to do that before putting in a pull request to [bcgov/envreportutils](http://github.com/bcgov/envreportutils) (if at all).

I'm officially ready for review, so let me know what you think!

stephhazlitt commented:
Thanks for your work on this so far, it is looking great. Unfortunately the pop-ups are not behaving for me (flickering, can really only see the labels)&mdash;I am wondering if this is related to the Java? I will see if I can troubleshoot tomorrow.

Some minor comments for discussion:
- I think we can remove the legend in the regional plots?
- While I cannot view the region pop-ups, one thought is to remove the plot titles (and number of wells) and shift that content into the pop-up itself, if you feel that the content is sparse?
- The bc silhouette could be more centered in @ateucher's `envreportutils::bc_home_button()`?
- What about maybe clipping the `bcmaps::nr_regions()` with `bcmaps::bc_bound()` to remove the ocean?

Overall I think it is really close to the vision outlined in the opportunity. @ateucher do you have any opinion re: location of various leaflet-related functions (`envreportutils` or analysis repos)? Tx both!

stephhazlitt commented:
Thanks for the tip, pop-ups rendering fine with line 135 commented out. The map looks great, imo.

_Regional Pop-Ups_: I think the regional pop-up content looks fine&mdash;relatively minimal as you say, but what is in there is informative and easy to digest. I think the bar chart titles are redundant, as the pop-up has the region title already. Maybe we could remove the title and shift the # wells info into a second line in the pop-up titles (e.g. Number of Wells Reported: 10)?

_Well Pop-Ups_: Looks like a bit more real-estate is needed for the monthly plots&mdash;the top, horizontal legend is cutting cut-off.  Let's be a bit more specific abt **More Info**, for example **Learn More About the Observation Well or Download the Data:**? I like the badge, although it does dominate the window, and I wonder if we have a bit of redundancy with reporting the trend category content generally? It is a lot of reporting content in one modal window, so really open to thoughts about what and how to present this for users.

ateucher commented:
For the little bc map being off-centre, I believe it's [this little bit of css](https://github.com/steffilazerte/envreportutils/blob/f61ff5a2d375e15047c5e119a0015be79c9c3c97/inst/css/wells-styles.css#L57-L59) that's pushing it down...

ateucher commented:
Looks really good Steffi! I also think clipping to the bc boundary would be nice. Also, I agree about giving the top-right hydrgraph a bit more emphasis, and the badge a little less...

steffilazerte commented:
Okay sounds good, I'll tweak the formatting and see what I can come up with!

ateucher commented:
As for the location of leaflet-related functions... I think the ones that are applicable to > 1 map and we can see using them for many other SoE maps (`popups_centre()`, `set_bc_view_on_close()`, `set_bc_view()`, possibly `popup_create_row()` and `popup_combine_rows()`) can stay in **envreportutils**. Could probably make the case for the caaqs one to be in there too since they are used in two indicators (and will be more in the future). The other ones (i.e., anything with  or `_groundwater` in the name can probably go in a `leaflet_functions.R` (or something) file in the analysis repo. I think the same goes for the css...?

steffilazerte commented:
Yeah, that definitely makes sense!

steffilazerte committed:
Move css and specific functions from envreportutils

steffilazerte committed:
Remove padding from top of home button

steffilazerte committed:
Clip natural resources polygon
- Clip NR bc map before making ggmap figures
- save for use in leaflet map
- adjust formating

steffilazerte committed:
Update to readr functions

steffilazerte committed:
Format regional popups
- Remove title and legend from figures
- Add sample size as subtitle

steffilazerte committed:
Use here() to specify source location

steffilazerte committed:
Format well popups
- Remove month plot
- Add more details
- Area plot larger text and minimize extra
- Increase map height

steffilazerte committed:
Correct/Clarify the number of wells as those with trend data

steffilazerte committed:
Make category names consistent through script and consistently title case (and sort wells by least to most decline)

stephhazlitt committed:
find functions

stephhazlitt committed:
functions now exported

ateucher commented:
This looks really nice, well done! One strange thing though - the links to the 'GW Network Map' in the well popups don't work. I get the hand cursor and the url shows up the browser status bar, but clicking doesn't do anything.

steffilazerte commented:
It's supposed to open in a new browser window. Is it possible it's being prevented by popup-blocking software?

steffilazerte committed:
Use keepInView as opposed to Java function to centre Popups

steffilazerte committed:
Merge branch 'update-2018' of github.com:bcgov/groundwater-levels-indicator into update-2018

stephhazlitt committed:
remove deprecated code; use percent_format()

stephhazlitt committed:
tweak labels

stephhazlitt committed:
add ow count; tweak labels

stephhazlitt committed:
comment

stephhazlitt committed:
remove deprecated code

stephhazlitt committed:
comment out month svgs

stephhazlitt committed:
rearrange code

stephhazlitt committed:
add licence header

stephhazlitt committed:
update text

ateucher committed:
Use nzchar; whitespace formatting

ateucher committed:
Whitespace formatting

ateucher committed:
Uncomment nLabeller definition

ateucher committed:
Whitespace formatting

ateucher committed:
Use names of colrs vector to set order

ateucher committed:
Fix loop to create wellMaps.

Because wellNum is an integer, using them for subsetting, wellMaps list was indexed by integer rather than name

ateucher committed:
Fix map_plot.
- put vars in a list in pmap
- convert names of wellMaps to integer for join

stephhazlitt committed:
move if(create_ggmaps)

stephhazlitt committed:
another style change

ateucher committed:
Add aquifer_id to leaflet map

stephhazlitt committed:
tweak y-axis breaks

stephhazlitt committed:
Merge branch 'update-2018' of https://github.com/bcgov/groundwater-levels-indicator into update-2018

stephhazlitt committed:
update to 2009

stephhazlitt committed:
add /data

stephhazlitt committed:
change geom_point colour

stephhazlitt committed:
update refs

ateucher committed:
Fix link to aquifer page if not known

ateucher committed:
Properly close <a> tags so links don't overflow into other divs

ateucher committed:
Add statement about insufficient wells in region popup

ateucher committed:
Show meta info in popup for wells without trend info

ateucher committed:
Stay silent when no aquifer info available

ateucher committed:
Remove commented code

ateucher committed:
Put insufficient statement on same line as number of wells analyzed

stephhazlitt committed:
save results_viz object; summary stats

stephhazlitt committed:
update gitignore

stephhazlitt committed:
working print ver w/o appendix

stephhazlitt committed:
start appendix refactor

stephhazlitt committed:
fix to work w/ list-columns

stephhazlitt committed:
remove appendix sandbox script

stephhazlitt committed:
add working appendix

stephhazlitt committed:
reduce PDF file size

ateucher committed:
Change bcmaps.rdata to bcmapsdata

ateucher committed:
Fix name of object that is changed and referenced elsewhere

ateucher committed:
Change pub month in pdf

ateucher committed:
Filter end date of data

ateucher committed:
Tweak run_all.R

ateucher committed:
Diff wells with convergence problems

ateucher committed:
create ggmaps

ateucher committed:
Make column names and position match previous results data

ateucher committed:
Round significance and percent missing

ateucher committed:
round years, rename CHEMISTRY_SITE_ID to EMS_ID so it is retained

ateucher committed:
Add caption to summary plot

ateucher committed:
Add caption to summary plot and table in print version

ateucher committed:
Add ghostscript argument dPrinted=false to preserve hyperlinks

ateucher committed:
Merge branch 'update-2018' of https://github.com/bcgov/groundwater-levels-indicator into update-2018

ateucher committed:
Text tweaks
